cli/azd/pkg/exec/command_runner.go

```diff
@@ -58,7 +58,14 @@ func (r *commandRunner) Run(ctx context.Context, args RunArgs) (RunResult, error
 
 	cmd.Dir = args.Cwd
 
-	var stdin, stdout, stderr bytes.Buffer
+	var stdin io.Reader
+	if args.StdIn != nil {
+		stdin = args.StdIn
+	} else {
+		stdin = new(bytes.Buffer)
+	}
+
+	var stdout, stderr bytes.Buffer
```

Comment:
I updated the default case to use the stdin, stdout & stderr passed on the constructor.  Not sure why/when we missed that earlier.  

Now it will use those values unless overridden on a particular call like we need for the `kubectl apply`